tcpdialer.go

```diff
@@ -271,6 +271,36 @@ func (d *TCPDialer) DialDualStackTimeout(addr string, timeout time.Duration) (ne
 	return d.dial(addr, true, timeout)
 }
 
+// FlushDNSCache clears all cached DNS entries, forcing fresh DNS lookups on subsequent dials.
+// This is useful when you want to ensure fresh DNS resolution, for example after network changes.
+func (d *TCPDialer) FlushDNSCache() {
+	d.tcpAddrsMap.Range(func(k, v any) bool {
+		d.tcpAddrsMap.Delete(k)
+		return true
+	})
+}
+
+// CleanDNSCache removes expired DNS cache entries based on DNSCacheDuration.
+// This is useful when you have set a longer DNSCacheDuration and want to manually
+// trigger cleanup of expired entries without waiting for the automatic cleanup.
+func (d *TCPDialer) CleanDNSCache() {
```

Comment:
Lets not expose this method. Expired DNS entries are already not used anymore. The only thing this method does is clean up some unused memory a bit earlier. I don't think that's worth increasing the API surface for.